### PR TITLE
schedulers: add switch for hot scheduler to forbid read or write scheduler

### LINKED ISSUE #4261: hot scheduler can forbid read or write scheduler
## Feature Request
Hot scheduler can forbid read or write scheduler using config
### Describe your feature request related problem
hot scheduler read and write will influnce each, in some case, we can turn off read or write scheduler to run hot scheduler. 

### Describe the feature you'd like
The read or write switch should configurable.

### PR DESCRIPTION
**Pull request by bufferflies**
Signed-off-by: bufferflies <1045931706@qq.com>

### What problem does this PR solve?
close  #4261 

### What is changed and how it works?
The read or write scheduler  depends on  forbidRWType switch 
### Check List

Tests

- Unit test
- Integration test

Code changes

- Has configuration change

Side effects

Related changes

### Release note

```release-note

add switch for hot scheduler to forbid read or write scheduler
```

### CODE REVIEW

File: server/schedulers/hot_region.go [hunk 1]
```diff
@@ -164,9 +164,14 @@ func (h *hotScheduler) dispatch(typ rwType, cluster opt.Cluster) []*operator.Ope
 
 	switch typ {
 	case read:
-		return h.balanceHotReadRegions(cluster)
+		if !h.conf.IsForbidRWType(typ) {
```
Comment (disksing) [inline server/schedulers/hot_region.go:167]: the if can move to before the switch.

**Comment by lhy1024** [comment]
I think we should consider hiding this configuration, it will only be used in the test

**Review by lhy1024 — COMMENTED** [review]
I think we should consider hiding this configuration, it will only be used in test

**Review by disksing — APPROVED** [review]

**Comment by bufferflies** [comment]
> I think we should consider hiding this configuration, it will only be used in test
yes, this switch just for test, so it will not show in config scheduler ,but will see it in config show all

**Review by lhy1024 — COMMENTED** [review]
the rest LGTM

File: server/schedulers/hot_region.go [hunk 2]
```diff
@@ -161,6 +161,9 @@ func (h *hotScheduler) dispatch(typ rwType, cluster opt.Cluster) []*operator.Ope
 	defer h.Unlock()
 
 	h.prepareForBalance(typ, cluster)
+	if h.conf.IsForbidRWType(typ) {
```
[the remark this answers is not in the document]
Comment (bufferflies) [reply server/schedulers/hot_region.go:164]: it will filter typ

**Review by lhy1024 — APPROVED** [review]

File: server/schedulers/hot_region.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (bufferflies) [reply server/schedulers/hot_region.go:165]: it should support to use api and metrics

**Comment by JmPotato** [comment]
Please address the commit with an associated issue.

**Comment by nolouch** [comment]
it fire a invalid-commit-message issue.

**Comment by rleungx** [comment]
> yes, this switch just for test, so it will not show in config scheduler ,but will see it in config show all

I think it won't be displayed in `config show all` either.

**Pull request merged**